### ISSUE TITLE
No more scrolling with iOS 15.1 

### ISSUE THREAD

**Issue by tifroz** [andreamazz/AMScrollingNavbar#406]
This can be reproduced with the demo app running `iOS 15.1` (screen capture)

https://user-images.githubusercontent.com/150367/139125055-b5a61d02-5b0f-40c2-9071-fdb729dc3d02.mov

**Comment by chinh-tran** [comment]
https://github.com/andreamazz/AMScrollingNavbar/blob/5016bc166af5cd08676fed5d762e00e6333a88ac/Source/ScrollingNavigationController.swift#L588
Apparently moving the navbar using the `frame` property no longer works with iOS 15.1 (https://developer.apple.com/forums/thread/691975).

An alternative is to use `CGAffineTransform`, which seems to fix the issue:

```swift
navigationBar.transform = CGAffineTransform(translationX: 0, y: navigationBar.transform.ty - delta)
```

**Comment by TheDreamsWind** [comment]
I'm experiencing similar problem under iOS 14.5, and as far as i'm concerned [the linked discussion](https://developer.apple.com/forums/thread/691975) has nothing to do with the library, since it modifies the nav. ber on behalf of the view controller owning it.

**Comment by guicolares** [comment]
Thanks @chinh-tran , I had the same bug and resolved it with this fix.  May I create a PR ? 

[my fix branch](https://github.com/guicolares/AMScrollingNavbar/commit/e863277474e211fc61a2fcb03d1f63a2eb1c0ae2)

**Comment by dongtg-1485** [comment]
@guicolares
Thanks for your pull request. But when you use `CGAffineTransform` to update `NavigationBar`, `topViewController.view.frame` was not updated when scroll slowly.
I think we need to find an other solution.

**Comment by closure11** [comment]
@andreamazz Do you have any ideas about this? Apparently the main feature of this library was broken.

**Comment by andreamazz** [comment]
> @andreamazz Do you have any ideas about this? Apparently the main feature of this library was broken.

Nope, I need to find the time to test this. Did anyone test against 15.2 beta by any chance?

**Comment by dungInte** [comment]
I have the same problem too. if use the transform by chinhtran it helps scroll to the top but navigation will lag frame size and title too.

**Comment by anhnt2309** [comment]
Any update on this? This break my app on app store.

**Comment by andreamazz** [comment]
I've looked into this, honestly I don't know how to fix it, you can see during the scrolling that UIKit is fighting against both the frame change and the opacity change, I'm afraid that's not much we can do at this point. I was hoping for an issue with 15.1, but 15.2.1 behaves the same. Why did they start enforcing this with 15.1 is beyond my comprehension.

**Comment by Okizeme** [comment]
I think that answer from Apple might interest you: https://developer.apple.com/forums/thread/691975
